### DISCUSSION TITLE
Sequencing workers in Horovod with Linear All reduce

### DISCUSSION THREAD

**Discussion by paridhika** [horovod/horovod#3759] [Q&A]
Hi,

I am running distributed training using Horovod+ Tensorflow and Linear allreduce on 4 workers connected through a switch. I observe from the traces that the communication from the three workers with the rank zero workers always happens in a fixed order. Workers transmit their gradients in sequence one after the other but I was expecting them to start transmission simultaneously. Can you please tell me if Horovod does some kind of orchestration to series their communication?

Thanks
Paridhika
